packages/core/src/util/txInspector.ts

```diff
@@ -24,35 +28,96 @@ export type StakeKeyRegistrationInspection = StakeAddressCertificate[];
 export type WithdrawalInspection = Lovelace;
 
 // Inspector types
+interface SentInspectorArgs {
+  addresses?: Address[];
+  stakeKeyHashes?: Ed25519KeyHash[];
+  rewardAccounts?: RewardAccount[];
+}
+export type SentInspector = (args: SentInspectorArgs) => Inspector<boolean>;
+export type TotalInputValuesInspector = (
+  ownAddresses: Address[],
+  historicalTxs: TxAlonzo[]
+) => Inspector<SendReceiveValueInspection>;
 export type SendReceiveValueInspector = (ownAddresses: Address[]) => Inspector<SendReceiveValueInspection>;
 export type DelegationInspector = Inspector<DelegationInspection>;
 export type StakeKeyRegistrationInspector = Inspector<StakeKeyRegistrationInspection>;
 export type WithdrawalInspector = Inspector<WithdrawalInspection>;
 
 /**
- * Inspects a transaction for value (coins + assets) sent by the provided addresses.
+ * Inspects a transaction for values (coins + assets) in inputs
+ * containing any of the provided addresses.
  *
  * @param {Address[]} ownAddresses own wallet's addresses
- * @returns {Value} total value sent
+ * @param {TxAlonzo[]} historicalTxs wallet's historical transactions
+ * @returns {Value} total value in inputs
  */
-export const valueSentInspector: SendReceiveValueInspector = (ownAddresses: Address[]) => (tx: TxAlonzo) => {
-  if (!isOutgoing(tx, ownAddresses)) return { coins: 0n };
-  const sentOutputs = tx.body.outputs.filter((out) => !isAddressWithin(ownAddresses)(out));
-  return coalesceValueQuantities(sentOutputs.map((output) => output.value));
+export const totalInputValuesInspector: TotalInputValuesInspector = (ownAddresses, historicalTxs) => (tx) => {
```

Comment:
How about this? Suggestions @rhyslbw ?
```suggestion
export const totalAddressInputsValueInspector: TotalAddressInputsValueInspector = (ownAddresses, historicalTxs) => (tx) => {
```